subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -881,10 +882,21 @@ int bt_mesh_scene_srv_set(struct bt_mesh_scene_srv *srv, uint16_t scene,
 		return -ENOENT;
 	}
 
+	curr = current_scene(srv);
+	if (scene == curr) {
+		srv->prev = scene;
+		srv->next = BT_MESH_SCENE_NONE;
+		srv->transition.time = 0;
+		srv->transition.delay = 0;
+		/* We're checking srv->next in the handler, so failure to cancel is okay: */
+		(void)k_work_cancel_delayable(&srv->work);
+		return 0;
```

Comment:
This is just a question: Why do we not call settings_load_subtree and scene_recall_complete when doing this?

Reply:
The state isn't actually changing, so we don't need to refresh it. If any of the affected models are out of sync with the previously set scene, they should have told the scene server, which would have invalidated the scene.